backend/app/policies/enrollment_policy.rb

```diff
@@ -15,17 +19,17 @@ def create?
   end
 
   def update?
+    if record.status_archived?
```

Comment:
this condition is not necessary as the condition `(record.status_draft? || record.status_changes_requested?)` below imply it